src/client/components/App/index.js

```diff
@@ -32,18 +23,31 @@ export default class App extends Component {
             <div>
                 <em>details: {this.props.info}</em>
                 <ErrorBoundary>
-                    { !this.state.clickedMovie ?
-                        <Search searchMovies={this.searchMovies}/> :
-                        <MovieDetails movie={this.state.clickedMovie}/>
+                    { !this.props.chosenMovie ?
+                        <Search/> :
+                        <MovieDetails movie={this.props.chosenMovie}/>
                     }
                 </ErrorBoundary>
 
                 <ErrorBoundary>
                     <ItemsList
-                        moviesList={this.state.searchResult}
                         handleMovieClick={this.handleMovieClick}/>
                 </ErrorBoundary>
             </div>
         );
     }
-}
\ No newline at end of file
+}
+
+export default connect(
+    (state) => {
+        return {
+            moviesList: state.moviesList,
+            chosenMovie: state.chosenMovie,
+        };
+    },
+    (dispatch) => {
+        return {
+            onShowMovie: (clickedMovie) => dispatch(showMovie(clickedMovie)),
+        }
+    },
```

Comment:
этот код можно упростить до

```suggestion
    { onShowMovie: showMovie },
```